services/databaseEncryptionService.ts

```diff
@@ -23,55 +22,73 @@ export class EncryptionError extends Error {
 export const ERROR_CODES = {
   USER_CANCELLED: 'USER_CANCELLED',
   KEY_NOT_FOUND: 'KEY_NOT_FOUND',
+  UNINITIALIZED_ENCRYPTION: 'UNINITIALIZED_ENCRYPTION',
 } as const;
 
-let keyCache: Uint8Array | null = null;
+let keyCache: string | null = null;
 let initializationPromise: Promise<void> | null = null;
+let reWrapPromise: Promise<void> | null = null;
 
-function generateRandomKey(): Uint8Array {
-  return Crypto.getRandomBytes(32);
+function generateRandomKeyHex(): string {
+  const bytes = Crypto.getRandomBytes(32);
+  return Array.from(bytes).map(b => b.toString(16).padStart(2, '0')).join('');
 }
 
+// expo-secure-store v15.0.7 throws ERR_AUTHENTICATION for both:
+// - Authentication required (key needs auth but requireAuthentication=false was used)
+// - User cancellation (user cancelled the biometric prompt)
+// Tested on Android. There's no reliable way to distinguish between these two scenarios.
+// We use the same check but keep separate functions for semantic clarity.
 function isCancellationError(error: unknown): boolean {
+  const code = (error as any)?.code;
+  return code === 'ERR_AUTHENTICATION';
+}
+
+function isAuthenticationRequiredError(error: unknown): boolean {
   if (!(error instanceof Error)) return false;
-  
-  const name = (error as any).name;
-  if (name === 'NotAllowedError' || name === 'ERR_CANCELED') return true;
-  
   const code = (error as any).code;
-  if (code === 'ERR_CANCELED' || code === 'USER_CANCELLED') return true;
-  
-  return error.message.includes('cancel');
+  return code === 'ERR_AUTHENTICATION';
 }
 
-function classifyError(error: unknown): never {
-  if (error instanceof EncryptionError) {
-    throw error;
-  }
-
-  if (isCancellationError(error)) {
-    throw new EncryptionError(ERROR_CODES.USER_CANCELLED, 'Authentication was cancelled');
-  }
 
-  throw error;
-}
-
-async function createNewKey(): Promise<Uint8Array> {
-  const key = generateRandomKey();
-  await SecureStore.setItemAsync(SECURE_STORE_KEYS.ENCRYPTION_KEY, base64.fromByteArray(key), { requireAuthentication: false });
+async function createNewKey(): Promise<string> {
+  const keyHex = generateRandomKeyHex();
+  await SecureStore.setItemAsync(SECURE_STORE_KEYS.ENCRYPTION_KEY, keyHex, { requireAuthentication: false });
   await setStoredEncryptionMode('basic');
```

Comment:
Like I mentioned in an earlier comment, you're making your life difficult with this concept of `basic`/`protected` mode. This could just be a boolean (e.g. `doesKeyAccessRequireAuthentication`).